### PR TITLE
fix duplicate names bug in ImporterContext

### PR DESCRIPTION
**Pull request by tdp2110**
Signed-off-by: Tom Peters <thomas.d.peters@gmail.com>

This is the fix I described from #466

### CODE REVIEW

**Comment by tdp2110** [comment]
Is it worth resolving conflicts on this PR or will this just be ignored?

**Comment by rmccorm4** [comment]
CC @kevinch-nv

**Comment by tdp2110** [comment]
FYI LLVM has to solve a similar problem. For example, see https://llvm.org/docs/tutorial/OCamlLangImpl3.html

> One nice thing about LLVM is that the name is just a hint. For instance, if the code above emits multiple “addtmp” variables, LLVM will automatically provide each one with an increasing, unique numeric suffix.

You can see their implementation of value naming [here](https://github.com/llvm/llvm-project/blob/master/llvm/lib/IR/ValueSymbolTable.cpp#L102), which in the case of name conflicts dispatches [here](https://github.com/llvm/llvm-project/blob/master/llvm/lib/IR/ValueSymbolTable.cpp#L43). The basic pattern of the uniquification matches what's done here: there's a map of all used names, and they loop with increasing integer suffixes until something isn't in the map. The particular implementation of the map is not a `std::set` as here, but the idea is the same.

File: ImporterContext.hpp [hunk 1]
```diff
@@ -216,6 +215,22 @@ class ImporterContext final : public IImporterContext
             return _opsets.at(domain);
         }
     }
+
+  private:
+    static std::string freshName(std::set<std::string> &namesSet,
```
Comment (kevinch-nv) [inline ImporterContext.hpp:220]: Prefer a more verbose function name, such as `generateUniqueName`

Comment (tdp2110) [reply ImporterContext.hpp]: 👍 done

**Comment by tdp2110** [comment]
I changed the impl since I last pushed this: now the `ImporterContext` holds a single integer suffix counter. In the case of repeated names in an onnx graph, you won't get the same n^2 search for suffixes anymore.

Also, I realize clang-format may have done something weird with the comments in this file. I used the [.clang-format from TensorRT](https://github.com/NVIDIA/TensorRT/blob/master/.clang-format) (there's no .clang-format in this repo). I tried multiple versions of clang-format (7,8,9) and they all did the comment-wrapping like this (sometimes clang-format version matters!). If this is a problem, I can manually edit these lines.

**Pull request merged**